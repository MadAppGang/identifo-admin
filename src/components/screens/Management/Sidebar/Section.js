import React from 'react';
import PropTypes from 'prop-types';
import { NavLink, withRouter } from 'react-router-dom';

const SidebarSection = ({ path, title, match }) => {
  const isActive = match.params.section === path.split('/')[2];

  return (
    <NavLink
      exact
      to={path}
      className="iap-management-sidebar__section"
      activeClassName="iap-management-sidebar__section--active"
    >
      {isActive && (
        <div className="iap-management-sidebar__indicator" />
      )}
      {title}
    </NavLink>
  );
};

SidebarSection.propTypes = {
  path: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      section: PropTypes.string,
    }),
  }).isRequired,
};

export { SidebarSection };

export default withRouter(SidebarSection);
