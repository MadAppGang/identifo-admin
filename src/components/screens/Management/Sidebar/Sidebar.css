.iap-management-sidebar {
  display: flex;
  flex-direction: column;
  width: 250px;
  min-height: calc(100vh - 80px); /* header height */
  padding: 50px 0;
  background-color: #fff;
  position: fixed;
}

.iap-management-sidebar__section {
  color: #000;
  margin: 7px 0;
  text-decoration: none;
  font-size: 16px;
  position: relative;
  display: flex;
  align-items: center;
  padding-left: 30px;
}

.iap-management-sidebar__section:hover {
  opacity: .6;
}

.iap-management-sidebar__section.iap-management-sidebar__section--active {
  color: #1B9CFC;
}

.iap-management-sidebar__section.iap-management-sidebar__section--active:hover {
  opacity: 1;
}
