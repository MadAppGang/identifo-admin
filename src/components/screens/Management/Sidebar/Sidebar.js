import React, { Component } from 'react';
import Section from './Section';
import * as icons from '~/components/icons/sidebar';

class Sidebar extends Component {
  constructor() {
    super();

    this.state = {
      sections: [
        {
          title: 'Getting started',
          path: '/management',
          exact: true,
          Icon: icons.GettingStartedIcon,
        },
        {
          title: 'Users',
          path: '/management/users',
          Icon: icons.UsersIcon,
        },
        {
          title: 'Account',
          path: '/management/account',
          Icon: icons.AccountIcon,
        },
        {
          title: 'Database',
          path: '/management/database',
          Icon: icons.DatabaseIcon,
        },
        {
          title: 'Applications',
          path: '/management/applications',
          Icon: icons.ApplicationsIcon,
        },
        {
          title: 'Login Types',
          path: '/management/settings',
          Icon: icons.PasswordlessLoginIcon,
        },
        {
          title: 'Email Integration',
          path: '/management/email_integration',
          Icon: icons.EmailIcon,
        },
        {
          title: 'OpenID Settings',
          path: '/management/openid',
          Icon: icons.OpenIDIcon,
        },
        {
          title: 'Multi-factor Auth',
          path: '/management/multi-factor_auth',
          Icon: icons.MultiFactorAuthIcon,
        },
        {
          title: 'Hosted Pages',
          path: '/management/hosted_pages',
          Icon: icons.HostedPagesIcon,
        },
      ],
    };
  }

  render() {
    const { sections } = this.state;

    return (
      <aside className="iap-management-sidebar">
        {sections.map(section => <Section key={section.title} {...section} />)}
      </aside>
    );
  }
}

export default Sidebar;
