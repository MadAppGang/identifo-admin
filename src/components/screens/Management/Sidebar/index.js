import React, { Component } from 'react';
import Section from './Section';
import * as iconComponents from '~/components/icons/SidebarIcon';
import './Sidebar.css';

class Sidebar extends Component {
  constructor() {
    super();

    this.state = {
      sections: [
        {
          title: 'Getting started',
          path: '/management',
          Icon: iconComponents.GettingStartedIcon,
        },
        {
          title: 'Users',
          path: '/management/users',
          Icon: iconComponents.UsersIcon,
        },
        {
          title: 'Users Database',
          path: '/management/database',
          Icon: iconComponents.DatabaseIcon,
        },
        {
          title: 'Applications',
          path: '/management/applications',
          Icon: iconComponents.ApplicationsIcon,
        },
        {
          title: 'Email Integration',
          path: '/management/email_integration',
          Icon: iconComponents.EmailIcon,
        },
        {
          title: 'Settings',
          path: '/management/settings',
          Icon: iconComponents.SettingsIcon,
        },
        {
          title: 'OpenID Settings',
          path: '/management/openid',
          Icon: iconComponents.OpenIDIcon,
        },
        {
          title: 'Social Login',
          path: '/management/social_login',
          Icon: iconComponents.SocialIcon,
        },
        {
          title: 'Multi-factor Auth',
          path: '/management/multi-factor_auth',
          Icon: iconComponents.MultiFactorAuthIcon,
        },
        {
          title: 'Passwordless Login',
          path: '/management/passwordless_login',
          Icon: iconComponents.PasswordlessLoginIcon,
        },
        {
          title: 'Hosted Pages',
          path: '/management/hosted_pages',
          Icon: iconComponents.HostedPagesIcon,
        },
      ],
    };
  }

  render() {
    const { sections } = this.state;

    return (
      <aside className="iap-management-sidebar">
        {sections.map(section => <Section key={section.title} {...section} />)}
      </aside>
    );
  }
}

export default Sidebar;
