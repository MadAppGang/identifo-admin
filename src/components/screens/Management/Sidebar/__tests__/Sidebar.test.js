import Sidebar from '..';

describe('<Sidebar />', () => {
  test('renders as expected', () => {
    expect(shallow(<Sidebar />)).toMatchSnapshot();
  });
});
