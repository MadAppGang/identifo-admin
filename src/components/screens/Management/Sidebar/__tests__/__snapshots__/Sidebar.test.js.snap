// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> renders as expected 1`] = `
<aside
  className="iap-management-sidebar"
>
  <withRouter(SidebarSection)
    key="Getting started"
    path="/management/getting_started"
    title="Getting started"
  />
  <withRouter(SidebarSection)
    key="Users"
    path="/management/users"
    title="Users"
  />
  <withRouter(SidebarSection)
    key="Users Database"
    path="/management/users_database"
    title="Users Database"
  />
  <withRouter(SidebarSection)
    key="Applications"
    path="/management/applications"
    title="Applications"
  />
  <withRouter(SidebarSection)
    key="Email Integration"
    path="/management/email_integrations"
    title="Email Integration"
  />
  <withRouter(SidebarSection)
    key="Settings"
    path="/management/settings"
    title="Settings"
  />
  <withRouter(SidebarSection)
    key="OpenID Settings"
    path="/management/openid_settings"
    title="OpenID Settings"
  />
  <withRouter(SidebarSection)
    key="Social Login"
    path="/management/social_login"
    title="Social Login"
  />
  <withRouter(SidebarSection)
    key="2FA"
    path="/management/2fa"
    title="2FA"
  />
  <withRouter(SidebarSection)
    key="Passwordless login"
    path="/management/passwordless_login"
    title="Passwordless login"
  />
  <withRouter(SidebarSection)
    key="Hosted Pages"
    path="/management/hosted_pages"
    title="Hosted Pages"
  />
</aside>
`;
